src/utils/dotnetUtils.ts

```diff
@@ -6,112 +6,114 @@
+
+        const templatesVersion: string | undefined = templatesInstalled ? ext.context.globalState.get(templatesKey) : undefined;
+        const cliFeedJson: cliFeedJsonResponse | undefined = await tryGetCliFeedJson();
+        if (!cliFeedJson) {
+            if (templatesInstalled) {
+                return;
+            } else {
+                throw new Error(localize('retryInternet', 'There was an error in retrieving the templates.  Recheck your internet connection and try again.'));
+            }
         }
-    }
-
-    export async function uninstallDotnetTemplates(outputChannel: OutputChannel): Promise<void> {
-        await validateDotnetInstalled();
 
-        const tempFolder: string = os.tmpdir();
+        let runtime: ProjectRuntime | undefined = await functionRuntimeUtils.tryGetLocalRuntimeVersion();
+        if (runtime === undefined) {
+            const picks: IAzureQuickPickItem<ProjectRuntime>[] = [
+                { label: ProjectRuntime.beta, description: '(.NET Core)', data: ProjectRuntime.beta },
+                { label: ProjectRuntime.one, description: '(.NET Framework)', data: ProjectRuntime.one }
+            ];
+            const options: QuickPickOptions = { placeHolder: localize('pickTemplateVersion', 'Select the template version to install') };
+            runtime = (await ext.ui.showQuickPick(picks, options)).data;
+        }
 
-        outputChannel.show();
-        outputChannel.appendLine(localize('uninstallFuncProject', 'Uninstalling "{0}"', projectPackageId));
-        await cpUtils.executeCommand(undefined, tempFolder, 'dotnet', 'new', '--uninstall', projectPackageId);
-        outputChannel.appendLine(localize('uninstallFuncs', 'Uninstalling "{0}"', functionsPackageId));
-        await cpUtils.executeCommand(undefined, tempFolder, 'dotnet', 'new', '--uninstall', functionsPackageId);
-        outputChannel.appendLine(localize('finishedUninstallingTemplates', 'Finished uninstalling Azure Functions templates.'));
-    }
+        const latestRelease: string = await tryGetTemplateVersionSetting(actionContext, cliFeedJson, runtime) || cliFeedJson.tags[getFeedRuntime(runtime)].release;
+        if (templatesVersion !== latestRelease) {
+            ext.outputChannel.show();
+            ext.outputChannel.appendLine(localize('updatingTemplates', 'Updating .NET templates for Azure Functions...'));
+            if (templatesInstalled) {
+                await uninstallTemplates();
+            }
 
-    export async function installDotnetTemplates(ui: IAzureUserInput, outputChannel: OutputChannel): Promise<void> {
-        await validateDotnetInstalled();
+            const tempFolder: string = path.join(os.tmpdir(), fsUtil.getRandomHexString());
+            await fse.ensureDir(tempFolder);
+            try {
+                ext.outputChannel.appendLine(localize('installFuncProject', 'Installing "{0}"', projectPackageId));
```

Comment:
This stuff all works without permission elevation?

Reply:
Yep. Tested on mac, windows, and linux